tpm2_pytss/types.py

```diff
@@ -744,13 +744,24 @@ class TPMA_MEMORY(int):
 
 
 class TPM_OBJECT(object):
-    def __init__(self, _cdata=None):
+    def __init__(self, _cdata=None, **kwargs):
 
         if _cdata is None:
             _cdata = ffi.new(f"{self.__class__.__name__} *")
 
         self._cdata = _cdata
 
+        tipe = ffi.typeof(_cdata)
+        if tipe.kind == "pointer":
+            tipe = tipe.item
+        fields = [x[0] for x in tipe.fields]
+        for k, v in kwargs.items():
+            if k not in fields:
+                raise AttributeError(
+                    f"{self.__class__.__name__} has no field by the name of {k}"
+                )
```

Comment:
Couldn't this just be a if key not in self._cdata?
Tests?
What do we map type to in a TPMT_PUBLIC since type is reserved? It looks like Python style guide is to use the name with an underscore, ie `type_`

FYI I have some work here similarly related:
https://github.com/williamcroberts/tpm2-pytss/pull/new/add-int-type-parsers

Reply:
key in _cdata doesn't work, at least not for me, and I want make sure that it's an actual struct/union field, not an attribute of the ffi object.
Tests are coming.
type isn't really reserved, so having an attribute called type is OK, it seems to be bad practice, but as long as it's only attributes of TPMS_ATTEST and TPMT_PUBLIC my opinion is that it's OK.
```
>>> pub = TPMT_PUBLIC(type=TPM2_ALG.ECC)
>>> print(pub.type)
35
```

Reply:
Test added now

Reply:
> key in _cdata doesn't work, at least not for me, and I want make sure that it's an actual struct/union field, not an attribute of the ffi object.
> Tests are coming.
> type isn't really reserved, so having an attribute called type is OK, it seems to be bad practice, but as long as it's only attributes of TPMS_ATTEST and TPMT_PUBLIC my opinion is that it's OK.
> 
> ```
> >>> pub = TPMT_PUBLIC(type=TPM2_ALG.ECC)
> >>> print(pub.type)
> 35
> ```

type isn't reserved in the sense that it's in pythons list of reserved words, however you will override `type()` locally. Which means if we ever need normal type() we can encounter some weirdness. In this particular case, since it's getting unpacked from kwargs it would be fine, but if things change it could cause weird issues.